corehq/apps/reports/standard/sms.py

```diff
@@ -757,8 +757,11 @@ def get_queryset(self):
         if event_status_filter:
             data = data.filter(event_status_filter)
 
-        if self.phone_number_filter:
-            data = data.filter(messagingsubevent__sms__phone_number__contains=self.phone_number_filter)
+        if self.phone_number_or_email_filter:
+            if self.phone_number_or_email_filter.isdigit():
```

Comment:
I know this is different than what I said before, but it may actually make sense to just always search both. That handles email segments that are purely numeric and makes the operation a lot clearer to the user. Alternatively, it could skip phone numbers if not numeric, but always search email.

Reply:
This makes sense. We'll await what product says, but in the mean time I changed it so it reflects your logic. https://github.com/dimagi/commcare-hq/pull/27794/commits/064065b59ff8cfc0c666dd0299bec9bda3180476 Let me know if there's a more elegant way to do this query.

Testing where `email`=`knguyen123@dimagi.com` and `phone2`=`12345`

## When querying `123`:
(both email and phone show up)
<img width="1122" alt="Screen Shot 2020-06-15 at 12 54 42 PM" src="https://user-images.githubusercontent.com/615126/84684975-73c2ce80-af07-11ea-992b-77801cdcbd49.png">


## When querying `n123`:
(only email)
<img width="1153" alt="Screen Shot 2020-06-15 at 12 54 54 PM" src="https://user-images.githubusercontent.com/615126/84684987-76bdbf00-af07-11ea-9ac9-379a1c200682.png">